### ISSUE TITLE
v1.4.3 broke builds for gems like Mailcatcher

### ISSUE THREAD

**Issue by henders** [sparklemotion/sqlite3-ruby#327]
With a dockerfile like:
```
FROM php:7.4.12-fpm
RUN apt-get update && apt-get install -qy ruby ruby-dev rubygems build-essential libsqlite3-dev
RUN gem install sqlite3 --version '1.4.3'
RUN gem install mailcatcher
```

It fails to build against the sqlite3 gem with the error:
```
docker build .
[+] Building 8.1s (6/7)
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                                        0.0s
 => => transferring dockerfile: 230B                                                                                                                                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                           0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                             0.0s
 => [internal] load metadata for docker.io/library/php:7.4.12-fpm                                                                                                                                                                                                                                                           0.5s
 => [1/4] FROM docker.io/library/php:7.4.12-fpm@sha256:07db4f537d7ea591cd9cecda712aed03ac1aaba8f243961c39605d605b07d411                                                                                                                                                                                                     0.0s
 => CACHED [2/4] RUN apt-get update && apt-get install -qy ruby ruby-dev rubygems build-essential libsqlite3-dev                                                                                                                                                                                                            0.0s
 => ERROR [3/4] RUN gem install sqlite3 --version '1.4.3'                                                                                                                                                                                                                                                                   7.5s
------
 > [3/4] RUN gem install sqlite3 --version '1.4.3':
#6 5.806 Building native extensions. This could take a while...
#6 7.430 ERROR:  Error installing sqlite3:
#6 7.430 	ERROR: Failed to build gem native extension.
#6 7.430
#6 7.430     current directory: /var/lib/gems/2.5.0/gems/sqlite3-1.4.3/ext/sqlite3
#6 7.430 /usr/bin/ruby2.5 -r ./siteconf20220614-8-19pbpfn.rb extconf.rb
#6 7.430 checking for sqlite3.h... yes
#6 7.430 checking for pthread_create() in -lpthread... yes
#6 7.430 checking for -ldl... yes
#6 7.430 checking for sqlite3_libversion_number() in -lsqlite3... yes
#6 7.430 checking for rb_proc_arity()... yes
#6 7.430 checking for rb_integer_pack()... yes
#6 7.430 checking for sqlite3_initialize()... yes
#6 7.430 checking for sqlite3_backup_init()... yes
#6 7.430 checking for sqlite3_column_database_name()... yes
#6 7.430 checking for sqlite3_enable_load_extension()... yes
#6 7.430 checking for sqlite3_load_extension()... yes
#6 7.430 checking for sqlite3_open_v2()... yes
#6 7.430 checking for sqlite3_prepare_v2()... yes
#6 7.430 checking for sqlite3_int64 in sqlite3.h... yes
#6 7.430 checking for sqlite3_uint64 in sqlite3.h... yes
#6 7.430 creating Makefile
#6 7.430
#6 7.430 current directory: /var/lib/gems/2.5.0/gems/sqlite3-1.4.3/ext/sqlite3
#6 7.430 make "DESTDIR=" clean
#6 7.430
#6 7.430 current directory: /var/lib/gems/2.5.0/gems/sqlite3-1.4.3/ext/sqlite3
#6 7.430 make "DESTDIR="
#6 7.430 compiling aggregator.c
#6 7.430 compiling backup.c
#6 7.430 compiling database.c
#6 7.430 database.c: In function 'rb_sqlite3_disable_quirk_mode':
#6 7.430 database.c:84:1: error: expected ';' before '}' token
#6 7.430  }
#6 7.430  ^
#6 7.430 database.c: In function 'exec_batch':
#6 7.430 database.c:748:57: warning: passing argument 3 of 'sqlite3_exec' from incompatible pointer type [-Wincompatible-pointer-types]
#6 7.430      status = sqlite3_exec(ctx->db, StringValuePtr(sql), hash_callback_function, callback_ary, &errMsg);
#6 7.430                                                          ^~~~~~~~~~~~~~~~~~~~~~
#6 7.430 In file included from ./sqlite3_ruby.h:25,
#6 7.430                  from database.c:1:
#6 7.430 /usr/include/sqlite3.h:406:9: note: expected 'int (*)(void *, int,  char **, char **)' but argument is of type 'int (*)(VALUE,  int,  char **, char **)' {aka 'int (*)(long unsigned int,  int,  char **, char **)'}
#6 7.430    int (*callback)(void*,int,char**,char**),  /* Callback function */
#6 7.430    ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#6 7.430 database.c:748:81: warning: passing argument 4 of 'sqlite3_exec' makes pointer from integer without a cast [-Wint-conversion]
#6 7.430      status = sqlite3_exec(ctx->db, StringValuePtr(sql), hash_callback_function, callback_ary, &errMsg);
#6 7.430                                                                                  ^~~~~~~~~~~~
#6 7.430 In file included from ./sqlite3_ruby.h:25,
#6 7.430                  from database.c:1:
#6 7.430 /usr/include/sqlite3.h:407:3: note: expected 'void *' but argument is of type 'VALUE' {aka 'long unsigned int'}
#6 7.430    void *,                                    /* 1st argument to callback */
#6 7.430    ^~~~~~
#6 7.430 database.c:750:57: warning: passing argument 3 of 'sqlite3_exec' from incompatible pointer type [-Wincompatible-pointer-types]
#6 7.430      status = sqlite3_exec(ctx->db, StringValuePtr(sql), regular_callback_function, callback_ary, &errMsg);
#6 7.430                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~
#6 7.430 In file included from ./sqlite3_ruby.h:25,
#6 7.430                  from database.c:1:
#6 7.430 /usr/include/sqlite3.h:406:9: note: expected 'int (*)(void *, int,  char **, char **)' but argument is of type 'int (*)(VALUE,  int,  char **, char **)' {aka 'int (*)(long unsigned int,  int,  char **, char **)'}
#6 7.430    int (*callback)(void*,int,char**,char**),  /* Callback function */
#6 7.430    ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#6 7.430 database.c:750:84: warning: passing argument 4 of 'sqlite3_exec' makes pointer from integer without a cast [-Wint-conversion]
#6 7.430      status = sqlite3_exec(ctx->db, StringValuePtr(sql), regular_callback_function, callback_ary, &errMsg);
#6 7.430                                                                                     ^~~~~~~~~~~~
#6 7.430 In file included from ./sqlite3_ruby.h:25,
#6 7.430                  from database.c:1:
#6 7.430 /usr/include/sqlite3.h:407:3: note: expected 'void *' but argument is of type 'VALUE' {aka 'long unsigned int'}
#6 7.430    void *,                                    /* 1st argument to callback */
#6 7.430    ^~~~~~
#6 7.430 make: *** [Makefile:243: database.o] Error 1
#6 7.430
#6 7.430 make failed, exit code 2
#6 7.430
#6 7.430 Gem files will remain installed in /var/lib/gems/2.5.0/gems/sqlite3-1.4.3 for inspection.
#6 7.430 Results logged to /var/lib/gems/2.5.0/extensions/aarch64-linux/2.5.0/sqlite3-1.4.3/gem_make.out
------
executor failed running [/bin/sh -c gem install sqlite3 --version '1.4.3']: exit code: 1
```

If you revert to previous version of this gem:
```
FROM php:7.4.12-fpm
RUN apt-get update && apt-get install -qy ruby ruby-dev rubygems build-essential libsqlite3-dev
RUN gem install sqlite3 --version '1.4.2'
RUN gem install mailcatcher
```

Then it works fine:
```
docker build .
[+] Building 1.2s (8/8) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                                                        0.0s
 => => transferring dockerfile: 230B                                                                                                                                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                           0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                             0.0s
 => [internal] load metadata for docker.io/library/php:7.4.12-fpm                                                                                                                                                                                                                                                           1.1s
 => [1/4] FROM docker.io/library/php:7.4.12-fpm@sha256:07db4f537d7ea591cd9cecda712aed03ac1aaba8f243961c39605d605b07d411                                                                                                                                                                                                     0.0s
 => CACHED [2/4] RUN apt-get update && apt-get install -qy ruby ruby-dev rubygems build-essential libsqlite3-dev                                                                                                                                                                                                            0.0s
 => CACHED [3/4] RUN gem install sqlite3 --version '1.4.2'                                                                                                                                                                                                                                                                  0.0s
 => CACHED [4/4] RUN gem install mailcatcher                                                                                                                                                                                                                                                                                0.0s
 => exporting to image                                                                                                                                                                                                                                                                                                      0.0s
 => => exporting layers                                                                                                                                                                                                                                                                                                     0.0s
 => => writing image sha256:185a8d45e3d6ff12c787c13c66995968158111f193f109d220c05cb341d99895
```

**Comment by henders** [comment]
Oh, it looks like @tenderlove just made a new version v1.4.4 which fixes, thanks!!!

**Comment by henders** [comment]
Marking as fixed thanks for new version 1.4.4
